# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/DrawTargetD2D1.cpp
# Commit: 9bcf4283ca00
# Full Hash: 9bcf4283ca003f8572392b8f111361c12eecde3d
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2019-03-31 09:00:52
# Regressor Bug: 1539026
# File Overlap Count: 1
# Description:
#   Bug 1539026 - Skia m74 Moz2D fixes. r=rhunt
# ==============================================================================

diff -r 01b0009b3f67 -r 9bcf4283ca00 gfx/2d/DrawTargetD2D1.cpp
--- a/gfx/2d/DrawTargetD2D1.cpp	Tue Mar 26 11:06:47 2019 -0400
+++ b/gfx/2d/DrawTargetD2D1.cpp	Tue Mar 26 11:06:56 2019 -0400
@@ -1940,11 +1940,7 @@
     }
 
     if (pat->mBegin == pat->mEnd) {
-      uint32_t stopCount = stops->mStopCollection->GetGradientStopCount();
-      vector<D2D1_GRADIENT_STOP> d2dStops(stopCount);
-      stops->mStopCollection->GetGradientStops(&d2dStops.front(), stopCount);
-      d2dStops.back().color.a *= aAlpha;
-      return GetSolidColorBrush(d2dStops.back().color);
+      return CreateTransparentBlackBrush();
     }
 
     mDC->CreateLinearGradientBrush(
@@ -1973,6 +1969,10 @@
       return CreateTransparentBlackBrush();
     }
 
+    if (pat->mCenter1 == pat->mCenter2 && pat->mRadius1 == pat->mRadius2) {
+      return CreateTransparentBlackBrush();
+    }
+
     // This will not be a complex radial gradient brush.
     mDC->CreateRadialGradientBrush(
         D2D1::RadialGradientBrushProperties(